Gini-iOS-SDK/GINIDocumentTaskManager.m

```diff
@@ -61,76 +61,127 @@ - (instancetype)initWithAPIManager:(GINIAPIManager *)apiManager {
-        GINIDocument *document = [GINIDocument documentFromAPIResponse:task.result withDocumentManager:self];
-        return document;
-    }];
-    return GINIhandleHTTPerrors(documentTask);
-}
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromImage:(UIImage *)image {
-    return [self createDocumentWithFilename:fileName fromImage:image cancellationToken:nil];
+- (BFTask *)createDocumentWithFilename:(NSString *)fileName
+                             fromImage:(UIImage *)image {
+    return [self createDocumentWithFilename:fileName
+                                  fromImage:image
+                                    docType:nil];
 }
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromImage:(UIImage *)image cancellationToken:(BFCancellationToken *)cancellationToken {
-    return [self createDocumentWithFilename:fileName fromData:UIImageJPEGRepresentation(image, 0.2) docType:nil cancellationToken:cancellationToken];
+- (BFTask *)createDocumentWithFilename:(NSString *)fileName
+                             fromImage:(UIImage *)image
+                               docType:(NSString *)docType {
+    return [self createDocumentWithFilename:fileName
+                                   fromData:UIImageJPEGRepresentation(image, 0.2)
+                                    docType:@"image/jpeg"
+                          cancellationToken:nil];
 }
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromImage:(UIImage *)image docType:(NSString *)docType {
-    return [self createDocumentWithFilename:fileName fromImage:image docType:docType cancellationToken:nil];
+- (BFTask *)createDocumentWithFilename:(NSString *)fileName
+                              fromData:(NSData *)data
+                               docType:(NSString *)docType {
+    return [self createDocumentWithFilename:fileName
+                                   fromData:data
+                                    docType:docType
+                          cancellationToken:nil];
 }
 
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName fromImage:(UIImage *)image docType:(NSString *)docType cancellationToken:(BFCancellationToken *)cancellationToken {
-    return [self createDocumentWithFilename:fileName fromData:UIImageJPEGRepresentation(image, 0.2) docType:docType cancellationToken:cancellationToken];
-}
+- (BFTask *)createDocumentWithFilename:(NSString *)fileName
+                              fromData:(NSData *)data
+                               docType:(NSString *)docType
+                     cancellationToken:(BFCancellationToken *)cancellationToken {
+    NSParameterAssert([fileName isKindOfClass:[NSString class]]);
+    NSParameterAssert([data isKindOfClass:[NSData class]]);
+    
+    BFTask *createTask = [[_apiManager uploadDocumentWithData:data
+                                                  contentType:@"image/jpeg"
```

Comment:
Are you sure it's always a jpeg?

Reply:
I added this method before creating the mimeType helper. Nice catch!